kai/pydantic_models.py

```diff
@@ -55,6 +58,31 @@ def separate_sections(document):
     return sections
 
 
+def get_error_nodes(node: Node):
+    def traverse_tree_for_errors(node: Node):
+        for n in node.children:
+            if n.type == "ERROR" or n.is_missing:
+                yield n
+            if n.has_error:
+                yield from traverse_tree_for_errors(n)
+
+    yield from traverse_tree_for_errors(node)
+
+
+def has_parse_error(language: str, content: str) -> bool:
+    try:
+        parser = get_parser(language)
+    except Exception:
+        return False
```

Comment:
Can we log here so we know if parsing passed because the parser couldn't be found?

Reply:
Added!